presto-main/src/main/java/com/facebook/presto/operator/PartitionedOutputOperator.java

```diff
@@ -317,6 +332,7 @@ public PagePartitioner(
             this.partitionConstants = requireNonNull(partitionConstants, "partitionConstants is null").stream()
                     .map(constant -> constant.map(NullableValue::asBlock))
                     .collect(toImmutableList());
+            this.replicateAny = requireNonNull(replication, "replication is null") == Replication.REPLICATE_NULLS_AND_ANY;
```

Comment:
Doesn't matter. By doing `==` you export business logic from `Enum` to user classes, which is more fragile design than centralized logic.